pkg/p2p/libp2p/connections_test.go

```diff
@@ -357,11 +357,11 @@ func TestTopologyNotifiee(t *testing.T) {
 	expectPeers(t, s2, overlay1)
 	expectPeersEventually(t, s1, overlay2)
 
-	// expect that n1 notifee called with s2 overlay
 	waitAddrSet(t, &n1connectedAddr, &mtx, overlay2)
+	waitAddrSet(t, &n2connectedAddr, &mtx, overlay1)
```

Comment:
The previous version of tests was specifically made for connect to happen on one side, and for disconnect to trigger both sides.

Reply:
this was the intended behavior. why:
1. when you connect, it could be for two reasons: (i) peer dials in - then the notification should indeed happen (ii) dial out - then it might not be needed to notify topology again. this can cause a deadlock right now in kademlia, since you might be connecting from kademlia, then once the `afterConnect` tries to notify kademlia, it might create a deadlock since the access to `knownPeers` field has its own mutex (and a lock is already acquired in the manage loop iterator). not sure about this. will review tomorrow morning in depth but generally my hunch is that we would still need to regulate the cases in which we would like to connect but not notify the topology (in the case that the topology initiated the dialup). in case of debugapi for example you _would_ notify topology.